subsys/logging/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+
+target_sources_ifdef(CONFIG_SYS_LOG 		zephyr PRIVATE
+	${CMAKE_CURRENT_LIST_DIR}/sys_log.c)
+target_sources_ifdef(CONFIG_KERNEL_EVENT_LOGGER zephyr PRIVATE
+	${CMAKE_CURRENT_LIST_DIR}/event_logger.c
+	${CMAKE_CURRENT_LIST_DIR}/kernel_event_logger.c
+	)
```

Comment:
FYI;

this is the latest and greatest in how this kind of code should be written:

https://github.com/zephyrproject-rtos/zephyr/pull/1107/commits/ca8e20c7f32abb92f82c92302705cd2275bac654

The idea behind the zephyr_<function> API is that there is a lot of build script code that has very simple intent:
"Add this source file to the build"
"Add this flag when building source code"

This kind of code doesn't need the concept of libraries to be expressed. So we wrap the CMake API 
and change it slightly to reduce boilerplate. 

E.g. 
- PRIVATE is assumed
- The file path being relative to the current CMakeLists.txt file is assumed.
- Targeting the zephyr library is assumed.

As you can see from the linked commit, the build script code becomes a lot more
clean when we can make these assumptions.

We still need the zephyr_library_* API for more complex stuff like when:

A set of source files needs to be built differently from other source files.
A set of source has a public API of include paths.

etc.